django/test/runner.py

```diff
@@ -635,7 +635,11 @@ def build_suite(self, test_labels=None, extra_tests=None, **kwargs):
                     print('Excluding test tag(s): %s.' % ', '.join(sorted(self.exclude_tags)))
             all_tests = filter_tests_by_tags(all_tests, self.tags, self.exclude_tags)
 
-        all_tests = reorder_tests(all_tests, self.reorder_by, self.reverse)
+        # Put the failures detected at load time first for quicker feedback.
+        # _FailedTest objects include things like test modules that couldn't be
+        # found or that couldn't be loaded due to syntax errors.
+        test_types = (unittest.loader._FailedTest, *self.reorder_by)
```

Comment:
Why not add `_FailedTest` to the `DiscoverRunner.reorder_by`? :thinking: 

This would allow for custom behavior in subclasses.

Reply:
> Why not add _FailedTest to the DiscoverRunner.reorder_by?

Because someone that's already customized `reorder_by` then won't get the benefit of this change. I also think that someone shouldn't have to know about this Django implementation detail (and `unittest`'s internals) when setting their own `reorder_by`. I think the two should be independent of each other.

(The reason I initially added the get method was to give someone the ability to suppress this change, although I didn't have any specific use cases in mind. I figured it didn't hurt to have the method, so why not provide it just in case.)

Reply:
> I think the two should be independent of each other.

Agreed.

 > (The reason I initially added the get method was to give someone the ability to suppress this change if someone really needed to, although I didn't have any specific use cases in mind. I figured it didn't hurt to have the method, so why not provide it just in case.)

Let's leave it as it is. We can always add a hook later when there is a real use case.

Reply:
> Let's leave it as it is. We can always add a hook later when there is a real use case.

Thanks. Yes, that's right.